### PR TITLE
fix: prevent division by zero in TrajectoryImitationLoss at final denoising step

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Summary

- In `TrajectoryImitationLoss.align_trajectory()`, the target computation divides by `(sigma_ - sigma)`. At the final denoising step, `sigma_` is set to 0 and `sigma` approaches 0, making the denominator near-zero or exactly zero. This produces `inf`/`nan` gradients that corrupt distillation training.
- Added `.clamp(min=1e-6)` to the denominator to prevent the division by zero while preserving the mathematical intent for all other steps.

## Details

The problematic line in `diffsynth/diffusion/loss.py`:

```python
target = (latents_ - inputs_shared["latents"]) / (sigma_ - sigma)
```

At the last step of the scheduler loop, `sigma_` is explicitly set to `0` when `progress_id + 1 >= len(pipe.scheduler.timesteps)`, and `sigma` (the current step's sigma) is already near zero at the end of the schedule. The difference `(sigma_ - sigma)` therefore approaches zero, causing the division to produce `inf` or `nan` values that propagate through the loss and corrupt gradient updates.

The fix clamps the denominator to a small positive value:

```python
target = (latents_ - inputs_shared["latents"]) / (sigma_ - sigma).clamp(min=1e-6)
```

## Test plan

- [ ] Verify distillation training runs to completion without `nan`/`inf` losses
- [ ] Confirm training metrics remain consistent with expected convergence behavior

### CODE REVIEW

File: diffsynth/diffusion/loss.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def align_trajectory(self, pipe: BasePipeline, timesteps_teacher, trajectory_tea
                 progress_id_teacher = torch.argmin((timesteps_teacher - pipe.scheduler.timesteps[progress_id + 1]).abs())
                 latents_ = trajectory_teacher[progress_id_teacher]
             
-            target = (latents_ - inputs_shared["latents"]) / (sigma_ - sigma)
+            target = (latents_ - inputs_shared["latents"]) / (sigma_ - sigma).clamp(min=1e-6)
```
[the remark this answers is not in the document]
Comment (Mr-Neutr0n) [reply diffsynth/diffusion/loss.py]: Good catch — you're right that `.clamp(min=1e-6)` flips the sign when `(sigma_ - sigma)` is negative, which is the normal case since sigmas decrease monotonically. This would invert the target and break training.

I've updated the fix to preserve the sign of the denominator:

```python
denom = sigma_ - sigma
denom = torch.sign(denom) * torch.clamp(denom.abs(), min=1e-6)
target = (latents_ - inputs_shared["latents"]) / denom
```

This prevents division by zero while keeping the correct sign regardless of sigma ordering.
